server/rpm/pbench-server.spec

```diff
@@ -0,0 +1,241 @@
+Name:           pbench-server
+Version:        
+%define gdist   none
+Release:        1
+Summary:        The pbench server bits
+
+License:        GPLv3+
+URL:            http://github.com/distributed-systems-analysis/pbench
+Source0:        pbench-server-%{version}.tar.gz
+Buildarch:      noarch
+
+# policycoreutils for semanage and restorecon - used in pbench-server-activate-create-results-dir
+Requires:       policycoreutils
+
+%if 0%{?fedora}
+Requires:  policycoreutils-python-utils, python3-boto3
+%endif
+
+%if 0%{?rhel} == 7
+# need python3 - install scl to get it
+# scl-utils is in the main repo, but the scl python3
+# collection du jour is in the scl repo which
+# is assumed to have been added already with:
+#    yum install yum-utils
+#    yum-config-manager --enable rhel-server-rhscl-7-rpms
+#
+# we can't do that in the spec file, because of the yum lock,
+# so they have to be done beforehand.
+Requires: scl-utils, rh-python36, policycoreutils-python
+%endif
+
+%if 0%{?rhel} == 8
+# no python3-elasticsearch??
+# Requires:  policycoreutils-python-utils, python3-elasticsearch
+Requires:  policycoreutils-python-utils
+%endif
+
+Requires: npm
+
+# installdir has to agree with the definition of install-dir in
+# pbench-server.cfg, but we can't go out and pluck it from there,
+# because we don't know where the config file is. Note that we omit
+# the initial / - it is added in every use below.  IMO, that's more
+# readable since it appear in the middle of the path in all cases,
+# *except* in the %files section (and one instance in the %post
+# and %postun sections).
+
+%define installdir opt/pbench-server
+%define static %{installdir}/html/static
+
+%if 0%{?fedora}
+%define __python python3
+%endif
+
+%if 0%{?rhel}
+%define __python python3
+%endif
+
+%description
+The pbench server scripts.
+
+%prep
+
+%setup
+
+%build
+
+%install
+rm -rf %{buildroot}
+
+mkdir -p %{buildroot}/%{installdir}
+(cd ./server;
+ %{__make} install DESTDIR=%{?buildroot}/%{installdir} INSTALL="%{__install} -p";
+ %{__make} install-build-artifacts DESTDIR=%{?buildroot}/%{installdir} INSTALL="%{__install} -p")
+
+(cd ./web-server;
+ %{__make} clone DESTDIR=%{?buildroot}/%{static} INSTALL="%{__install} -p")
+
+cp -a %{?buildroot}/%{static}/package.json %{?buildroot}/%{installdir}
+
+# clean up unneeded files
+cd %{buildroot}/%{installdir}
+rm -rf Makefile html/static/Makefile \
+   lib/python3.7/site-packages/pbench/agent \
```

Comment:
Should this be `server` instead of `agent`?

Reply:
No - basically running `setup.py` installs a bunch of extra things and this code gets rid of them - hack, cough.